model/project.go

```diff
@@ -489,16 +490,17 @@ type TaskGroup struct {
 	Name string `yaml:"name" bson:"name"`
 
 	// data about the task group
-	MaxHosts              int             `yaml:"max_hosts" bson:"max_hosts"`
-	SetupGroupFailTask    bool            `yaml:"setup_group_can_fail_task" bson:"setup_group_can_fail_task"`
-	SetupGroupTimeoutSecs int             `yaml:"setup_group_timeout_secs" bson:"setup_group_timeout_secs"`
-	SetupGroup            *YAMLCommandSet `yaml:"setup_group" bson:"setup_group"`
-	TeardownGroup         *YAMLCommandSet `yaml:"teardown_group" bson:"teardown_group"`
-	SetupTask             *YAMLCommandSet `yaml:"setup_task" bson:"setup_task"`
-	TeardownTask          *YAMLCommandSet `yaml:"teardown_task" bson:"teardown_task"`
-	Timeout               *YAMLCommandSet `yaml:"timeout,omitempty" bson:"timeout"`
-	Tasks                 []string        `yaml:"tasks" bson:"tasks"`
-	Tags                  []string        `yaml:"tags,omitempty" bson:"tags"`
+	MaxHosts                int             `yaml:"max_hosts" bson:"max_hosts"`
+	SetupGroupFailTask      bool            `yaml:"setup_group_can_fail_task" bson:"setup_group_can_fail_task"`
+	TeardownTaskCanFailTask bool            `yaml:"teardown_task_can_fail_task" bson:"teardown_task_can_fail_task"`
```

Comment:
nit: can you keep the setupgroup variables together and move this before or after (or next to teardown)